server/src/clientglue.js

```diff
@@ -1,30 +1,87 @@
-/* globals window, document */
+/* jshint browser:true */
 
 let React = require("react"),
-  url = require("url"),
   { FrameFactory } = require("./views/frame.js"),
   { setGitRevision, staticLink } = require("./linker"),
   { AbstractShot } = require("../../shared/dist/shot");
 
-function clientGlue(data) {
-  setGitRevision(data.gitRevision);
-  let shot = new AbstractShot(data.backend, data.id, data.shot);
-  let frame = FrameFactory({
-    staticLink: staticLink,
-    backend: data.backend,
-    shot: shot,
-    id: data.id,
-    shotDomain: data.shotDomain,
-    query: data.query,
-    params: data.params
-  });
+// This represents the model we are rendering:
+let model;
+
+exports.setModel = function (data) {
+  let firstSet = ! model;
+  model = data;
+  model.shot = new AbstractShot(data.backend, data.id, data.shot);
+  render();
+  if (firstSet) {
+    window.addEventListener("hashchange", refreshHash, false);
+    refreshHash();
+  }
+};
+
+function render() {
+  setGitRevision(model.gitRevision);
+  let attrs = {
+    staticLink: staticLink
+  };
+  for (let attr in model) {
+    attrs[attr] = model[attr];
+  }
```

Comment:
Neat, I didn't know you could do `{...model}`. Is that a JSX extension?

`{staticLink, ...model}` is even more compact. :wink: